pkg/utils/server.go

```diff
@@ -150,3 +152,47 @@ func OpiAdressFamilyToSpdk(adrfam pb.NvmeAddressFamily) string {
 
 	return strings.ReplaceAll(adrfam.String(), "NVME_ADDRESS_FAMILY_", "")
 }
+
+// SpdkAddressFamilyToOpi converts a SPDK address family string to the corresponding OPI NvmeAddressFamily enum.
+func SpdkAddressFamilyToOpi(adrfam string) pb.NvmeAddressFamily {
+	switch strings.ToLower(adrfam) {
+	case "ipv4":
+		return pb.NvmeAddressFamily_NVME_ADDRESS_FAMILY_IPV4
+	case "ipv6":
+		return pb.NvmeAddressFamily_NVME_ADDRESS_FAMILY_IPV6
+	case "ib":
+		return pb.NvmeAddressFamily_NVME_ADDRESS_FAMILY_IB
+	case "fc":
+		return pb.NvmeAddressFamily_NVME_ADDRESS_FAMILY_FC
+	case "intra_host":
+		return pb.NvmeAddressFamily_NVME_ADDRESS_FAMILY_INTRA_HOST
+	default:
+		return pb.NvmeAddressFamily_NVME_ADDRESS_FAMILY_UNSPECIFIED
+	}
+}
+
+// ParseOpiTransportType parses a string to the corresponding OPI TransportType enum.
+func ParseOpiTransportType(t string) pb.NvmeTransportType {
+	switch strings.ToLower(t) {
+	case "fc":
+		return pb.NvmeTransportType_NVME_TRANSPORT_TYPE_FC
+	case "pcie":
+		return pb.NvmeTransportType_NVME_TRANSPORT_TYPE_PCIE
+	case "rdma":
+		return pb.NvmeTransportType_NVME_TRANSPORT_TYPE_RDMA
+	case "tcp":
+		return pb.NvmeTransportType_NVME_TRANSPORT_TYPE_TCP
+	default:
+		return pb.NvmeTransportType_NVME_TRANSPORT_TYPE_UNSPECIFIED
+	}
+}
+
+// ParseInt64 parses a string into int64, returning -1 on failure.
+func ParseInt64(s string) int64 {
```

Comment:
would it make sense to name it `ParseTrsvcid`? Otherwise it is implied that we can use this function to parse any string to int64, but in fact we cannot parse `-1`